### PR TITLE
Remove theme

### PR DESCRIPTION
**Pull request by j0sephh123**
Remove attempts for theming altogether.

### CODE REVIEW

File: src/store/mutations.js [hunk 1]
```diff
@@ -25,15 +25,11 @@ export const mutations= {
...
-    //   text,
-    // }
   },
   setOneStock(state, oneStock) {
     state.oneStock = oneStock;
   },
+  setCurrentTheme(state, themeString) {
+    state.theme.current = themeString;
+  }
 }
```
Comment (momozor) [inline src/store/mutations.js:35]:
```suggestion
}

```

File: src/store/actions.js [hunk 2]
```diff
@@ -69,7 +68,7 @@ export const actions = {
   },
   async getOneStock({ commit }, { stock, id }) {
     let item = await api.getOne(stock, id);
-    console.log(item);
+    // console.log(item);
```
Comment (momozor) [inline src/store/actions.js:71]:
```suggestion
```

File: src/store/actions.js [hunk 3]
```diff
@@ -17,16 +19,13 @@ export const actions = {
     // but it finally works
   },
   async submit({ state, dispatch, commit }, fields) {
-    // parse the data for the different types of stocks
-    let submitData = InputParser.submitData(fields, state.activeTab);
+    console.log(state.activeTab, fields);
+    let stock = stockFactory(state.activeTab, fields);
```
Comment (momozor) [inline src/store/actions.js:23]:
```suggestion
    const stock = stockFactory(state.activeTab, fields);
```

File: src/App.vue [hunk 4]
```diff
@@ -59,10 +56,7 @@ export default {
       updateObject: 'updateObject',
     }),
   },
-  methods: {
-  }
+  methods: {}
```
Comment (momozor) [inline src/App.vue:59]:
```suggestion
```

If there is no usage, better to delete this.

File: src/Index.vue [hunk 5]
```diff
@@ -15,4 +15,8 @@ export default {
     NavComponent: () => import("./components/Nav")
   }
 };
-</script>
\ No newline at end of file
+</script>
+
+<style lang="scss">
+@import "./styles/app.scss";
+</style>
```
Comment (momozor) [inline src/Index.vue:22]:
```suggestion
</style>

```

File: src/classes/Api.js [hunk 6]
```diff
@@ -1,6 +1,7 @@
 /*eslint-disable*/
 import axios from "axios";
 import router from "../router";
+import { statusCode } from '../constants';
```
Comment (momozor) [inline src/classes/Api.js:4]:
```suggestion
import { statusCode } from "../constants";
```

File: src/classes/Api.js [hunk 7]
```diff
@@ -61,14 +70,43 @@ class API {
 
   async resetDatabase() {
     let { status } = (await axiosInstance.get("list/reset-database")).data;
```
Comment (momozor) [inline src/classes/Api.js:72]:
```suggestion
    const status = (await axiosInstance.get("list/reset-database")).data;
```
Is it a requirement to use `{ var }` in this case?

File: src/classes/StockModel.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  fields['stockCategory'] = type;
+  
+  if (type === 'food') {
+    return new Food(fields);
+  } else if (type === 'water') {
+    return new Water(fields);
+  }
+
+  return new Stock(fields);
+}
```
Comment (momozor) [inline src/classes/StockModel.js:51]:
```suggestion
}

```

File: src/components/Table.vue [hunk 9]
```diff
@@ -62,7 +62,4 @@ export default {
   updated() {
   }
 };
-</script>
-
-<style>
-</style>
\ No newline at end of file
+</script>
```
Comment (momozor) [inline src/components/Table.vue:65]:
```suggestion
</script>

```

File: src/components/Table.vue [hunk 10]
```diff
@@ -62,7 +62,4 @@ export default {
   updated() {
```
Comment (momozor) [inline src/components/Table.vue:62]:
```suggestion
```

File: src/components/Table.vue [hunk 11]
```diff
@@ -62,7 +62,4 @@ export default {
...
   }
```
Comment (momozor) [inline src/components/Table.vue:63]:
```suggestion
```

File: src/components/UpdateModal.vue [hunk 12]
```diff
@@ -42,46 +45,16 @@ export default {
...
-  margin: 0px auto;
-  padding: 20px 30px;
-  background-color: #fff;
-  border-radius: 4px;
-  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.33);
-  transition: all 0.3s ease;
-}
-</style>
\ No newline at end of file
+</script>
```
Comment (momozor) [inline src/components/UpdateModal.vue:60]:
```suggestion
</script>

```

File: src/store/actions.js [hunk 13]
```diff
@@ -17,16 +19,13 @@ export const actions = {
...
 
-    let result = await api.create(submitData);
+    let result = await api.create(stock);
```
Comment (momozor) [inline src/store/actions.js:25]:
```suggestion
    const result = await api.create(stock);
```
If you are going to change a variable directly later in a code block, use `let`. Else, use `const`.

In Javascript, as far as I know, it let you mutate an const declared variable/object ***properties***, but not the variable/object itself.

If you want to make both the variable/object and the properties immutable, see - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze

File: src/store/actions.js [hunk 14]
```diff
@@ -17,16 +19,13 @@ export const actions = {
...
 
-    // for now we get only result.status, so, we fetch the whole
-    // dataset again in the future probably just get the newly created
-    // stock and add it do the existing stocks
-    if (result.status === 200) {
-      let exists = result.data.status === 404;
+    if (result.status === statusCode.success) {
+      let exists = result.data.status === statusCode.notFound;
```
Comment (momozor) [inline src/store/actions.js:28]:
```suggestion
      const exists = result.data.status === statusCode.notFound;
```
If you are going to change a variable directly later in a code block, use `let`. Else, use `const`.

In Javascript, as far as I know, it let you mutate an `const` declared variable/object ***properties***, but not the variable/object itself.

If you want to make both the variable/object and the properties immutable, see - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze

File: src/store/actions.js [hunk 15]
```diff
@@ -40,7 +39,7 @@ export const actions = {
     let result = await api.remove(id, state.activeTab);
```
Comment (momozor) [inline src/store/actions.js:39]:
```suggestion
    const result = await api.remove(id, state.activeTab);
```
If you are going to change a variable directly later in a code block, use `let`. Else, use `const`.

In Javascript, as far as I know, it let you mutate an `const` declared variable/object ***properties***, but not the variable/object itself.

If you want to make both the variable/object and the properties immutable, see - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze

File: src/store/actions.js [hunk 16]
```diff
@@ -52,7 +51,7 @@ export const actions = {
 
     let result = await api.update(stock);
```
Comment (momozor) [inline src/store/actions.js:52]:
```suggestion
    const result = await api.update(stock);
```
If you are going to change a variable directly later in a code block, use `let`. Else, use `const`.

In Javascript, as far as I know, it let you mutate an `const` declared variable/object ***properties***, but not the variable/object itself.

If you want to make both the variable/object and the properties immutable, see - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze

File: src/tests/unit/components/sidebar.spec.js [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+  });
+
+  test("active slug corresponds to active sidebar item", () => {
+    expect(
+      wrapper.get(`.parent_box router-link-stub`).attributes('to')
+    )
+    .toBe(
+      path
+    );
+
```
Comment (momozor) [inline src/tests/unit/components/sidebar.spec.js:51]:
```suggestion
```

File: src/tests/unit/components/sidebar.spec.js [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+  });
+
+  test("sidebar toggles on click", async () => {
+    let beforeClick = wrapper.vm.$data.toggled;
```
Comment (momozor) [inline src/tests/unit/components/sidebar.spec.js:55]:
```suggestion
    const beforeClick = wrapper.vm.$data.toggled;
```
If you are going to change a variable directly later in a code block, use `let`. Else, use `const`.

In Javascript, as far as I know, it let you mutate an `const` declared variable/object ***properties***, but not the variable/object itself.

If you want to make both the variable/object and the properties immutable, see - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze

File: src/tests/unit/components/sidebar.spec.js [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+    
+    await wrapper.get(".fa-bars").trigger('click');
+
+    let afterClick = wrapper.vm.$data.toggled;
```
Comment (momozor) [inline src/tests/unit/components/sidebar.spec.js:59]:
```suggestion
    const afterClick = wrapper.vm.$data.toggled;
```
If you are going to change a variable directly later in a code block, use `let`. Else, use `const`.

In Javascript, as far as I know, it let you mutate an `const` declared variable/object ***properties***, but not the variable/object itself.

If you want to make both the variable/object and the properties immutable, see - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze

File: src/views/About.vue [hunk 20]
```diff
@@ -17,8 +17,4 @@
 export default {
 
 }
-</script>
-
-<style>
-
-</style>
\ No newline at end of file
+</script>
```
Comment (momozor) [inline src/views/About.vue:20]:
```suggestion
</script>

```

File: src/views/Charts.vue [hunk 21]
```diff
@@ -0,0 +1,14 @@
...
+</template>
+
+<script>
+
+export default {
+  components: {
+    ChartComponent: () => import ('../components/Chart'),
+  },
+}
+</script>
```
Comment (momozor) [inline src/views/Charts.vue:14]:
```suggestion
</script>

```

File: src/views/Settings.vue [hunk 22]
```diff
@@ -9,40 +9,24 @@
...
+export default {
+  computed: {
+    ...mapGetters({
+      theme: 'theme',
+    }),
+  },
 }
-</style>
\ No newline at end of file
+</script>
```
Comment (momozor) [inline src/views/Settings.vue:32]:
```suggestion
</script>

```

File: src/views/Stock.vue [hunk 23]
```diff
@@ -90,11 +105,43 @@ export default {
...
+      const { stock, id } = this.$router.currentRoute.params;
+      this.id = id;
+      this.stockType = stock;
+      this.$store.dispatch("getOneStock", { stock, id });
+      
+      if(loadAll) {
+        this.allStocks = (await api.get(this.stockType)).data.stocks;
+      } 
+      
+      const currentStockIndex = this.allStocks.findIndex(item => item.id == +this.id)
```
Comment (momozor) [inline src/views/Stock.vue:121]:
```suggestion
      const currentStockIndex = this.allStocks.findIndex(item => item.id === +this.id) 
```

File: src/views/Stock.vue [hunk 24]
```diff
@@ -106,87 +153,20 @@ export default {
...
-.arrows .fa-angle-left:hover,
-.arrows .fa-angle-right:hover {
-  background: var(--primary);
-  color: white;
-  border-radius: 8px;
-  padding: 2px 8px;
-}
-</style>
\ No newline at end of file
+</script>
```
Comment (momozor) [inline src/views/Stock.vue:172]:
```suggestion
</script>

```

File: src/views/Survival.vue [hunk 25]
```diff
@@ -1,33 +1,45 @@
...
+// }
+
+// {
+//     "totalSurvivalDays": 1,
+//     "status": 200,
+//     "survivalAlertType": "warning" | "info" "success"
+// }
 
-<style>
+/**
```
Comment (momozor) [inline src/views/Survival.vue:30]:
```suggestion
```

File: src/views/Survival.vue [hunk 26]
```diff
@@ -1,33 +1,45 @@
...
+ *
```
Comment (momozor) [inline src/views/Survival.vue:31]:
```suggestion
```

File: src/views/Survival.vue [hunk 27]
```diff
@@ -1,33 +1,45 @@
...
+// {
+//     "totalSurvivalDays": 1,
+//     "status": 200,
+//     "survivalAlertType": "warning" | "info" "success"
+// }
 
-<style>
+/**
+ * 
+ */
```
Comment (momozor) [inline src/views/Survival.vue:32]:
```suggestion
```

File: src/views/Survival.vue [hunk 28]
```diff
@@ -1,33 +1,45 @@
...
+    // alertClassName: enum [success, info, warning, danger]
+    activeAlert: null,
+  }),
+  async mounted() {
+    
+    this.activeAlert = await api.getSurvival();
+    console.log(this.activeAlert);
+  },
+}
+</script>
```
Comment (momozor) [inline src/views/Survival.vue:45]:
```suggestion
</script>

```

**Comment by momozor** [comment]
I really should do `apply multiple suggestions batch` next time. Didn't know each applied suggestion equals to a single commit hash.

**Comment by momozor** [comment]
@j0sephh123  Anyway, there are few suggestions left to be applied. I didn't dare to do those on my own since I don't fully understand them and they are not automatically tested.

**Comment by j0sephh123** [comment]
I will now look at the suggestions and comment on that.

File: src/classes/Api.js (hunk 7 above)
Comment (j0sephh123) [reply src/classes/Api.js:72]: Nah, it's fine in both cases, just shorter syntax.

Comment (j0sephh123) [reply src/classes/Api.js:72]: `let { status } = (await axiosInstance.get("list/reset-database")).data;`
is actually equal to 
`const status = (await axiosInstance.get("list/reset-database")).data.status;`

Comment (momozor) [reply src/classes/Api.js:72]: Alright. I'll use you suggestion.

**Comment by j0sephh123** [comment]
Yes, by default `const` should be used.

**Pull request merged**